### PR TITLE
feat: implement websockets for ethereum and improve type naming consistency

### PR DESCRIPTION
**Pull request by kaladinlight**
- First cut of websocket in the eth chain adapter
- Bitcoin stubbed out as not implemented
- Updated import/export aliases (namespacing) to improve readability
- Removed unnecessary COINPrefix for classes and types already contained within a namespaced coin

### CODE REVIEW

File: packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.test.ts [hunk 1]
```diff
@@ -68,22 +69,22 @@ describe('BitcoinChainAdapter', () => {
     })
   })
```
Comment (mvpratt) [inline packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.test.ts:71]: are these comments intentional?

Comment (mvpratt) [reply packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.test.ts:71]: 👍

File: package.json [hunk 2]
```diff
@@ -31,6 +31,7 @@
     "@semantic-release/changelog": "^5.0.1",
     "@semantic-release/git": "^9.0.0",
     "@types/jest": "^27.0.1",
+    "@types/node": "^16.11.0",
```
[the remark this answers is not in the document]
Comment (kaladinlight) [reply package.json:34]: Thoughts on engine version in base package.json?

File: packages/types/src/chain-adapters/ethereum.ts [hunk 3]
```diff
@@ -2,18 +2,25 @@ import { ContractTypes } from '../base'
...
   contract: string
   precision: number
   name: string
   symbol: string
   contractType: ContractTypes
 }
 
+export type TokenWithBalance = Token & {
+  balance: string
+}
```
[the remark this answers is not in the document]
Comment (kaladinlight) [reply packages/types/src/chain-adapters/ethereum.ts:18]: Awesome, make sense!

File: packages/chain-adapters/src/ethereum/EthereumChainAdapter.ts [hunk 4]
```diff
@@ -250,9 +236,62 @@ export class EthereumChainAdapter implements ChainAdapter<ChainTypes.Ethereum> {
...
-    return { valid: false, result: ChainAdapters.ValidAddressResultType.Invalid }
+    if (isValidAddress) return { valid: true, result: chainAdapters.ValidAddressResultType.Valid }
+    return { valid: false, result: chainAdapters.ValidAddressResultType.Invalid }
+  }
+
+  async subscribeTxs(
+    input: chainAdapters.SubscribeTxsInput,
+    onMessage: (msg: chainAdapters.SubscribeTxsMessage<ChainTypes.Ethereum>) => void,
+    onError: (err: chainAdapters.SubscribeError) => void
+  ): Promise<void> {
```
[the remark this answers is not in the document]
Comment (kaladinlight) [reply packages/chain-adapters/src/ethereum/EthereumChainAdapter.ts:249]: Def would be down to talk about that with you!

File: packages/chain-adapters/src/ChainAdapterManager.ts [hunk 5]
```diff
@@ -14,17 +18,26 @@ export class ChainAdapterManager {
     if (!unchainedUrls) {
       throw new Error('Blockchain urls required')
     }
-    ;(Object.entries(unchainedUrls) as Array<[keyof UnchainedUrls, string]>).forEach(
-      ([type, basePath]) => {
+    ;(Object.entries(unchainedUrls) as Array<[keyof UnchainedUrls, UnchainedUrl]>).forEach(
+      ([type, { httpUrl: basePath, wsUrl }]) => {
```
[the remark this answers is not in the document]
Comment (reidrankin) [reply packages/chain-adapters/src/ChainAdapterManager.ts]: not that, just more that if `wsUrl` is a name `httpUrl` should be too. besides, `basePath` really only has meaning in the context of the `Configuration()` calls.

File: packages/swapper/src/swappers/zrx/approvalNeeded/approvalNeeded.ts [hunk 6]
```diff
@@ -35,7 +34,7 @@ export async function approvalNeeded(
     throw new SwapError('ZrxSwapper:approvalNeeded only Ethereum chain type is supported')
   }
 
-  const adapter: ChainAdapter<ChainTypes> = adapterManager.byChain(sellAsset.chain)
+  const adapter = adapterManager.byChain<ChainTypes>(sellAsset.chain)
```
[the remark this answers is not in the document]
Comment (reidrankin) [reply packages/swapper/src/swappers/zrx/approvalNeeded/approvalNeeded.ts:37]: it works fine with the inferred parameter type
```suggestion
  const adapter = adapterManager.byChain(sellAsset.chain)
```

**Pull request merged**